engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableAggregationTest.java

```diff
@@ -4090,6 +4090,40 @@ public void testIssue5820() throws ExecutionException, InterruptedException, Tim
         assertTableEquals(expected, result);
     }
 
+    @Test
+    public void testDH20997() {
+        final QueryTable table = testRefreshingTable(i(0).toTracking(), intCol("x", 0));
+        final Table distinct = table.aggBy(AggDistinct("x")).ungroup();
+        final Table distinctWithValues = distinct.update("extra=`value_string`");
+
+        final Table result = table.naturalJoin(distinctWithValues, "x");
```

Comment:
Since this seems to be related to previous values, should we also add an update validator onto the distinct table?  Would that also catch the bug, and maybe do it a bit more directly?